src/interface.jl

```diff
@@ -1,4 +1,10 @@
 
+abstract type AbstractCalculator end
+
+struct DummyState end
+calculator_state(::AbstractCalculator) = DummyState()
+update_state(::Any, ::DummyState, ::Any, ::Any) = DummyState()
```

Comment:
I would use

```julia
calculator_internal_state(::Any) = missing

update_calculator_state(
    update_algorithm::Any,
    initial_internal_state::Missing,
    initial_state::Any,
    new_state::Any
) = missing    # return new internal state
```

I would also add documentation on why to do this.

Also, `update_algorithm` is potentially very problematic. This stems from where are you going to define it. If it is defined for individual packages (like DFTK) then it should not be in the interface, or it will cause problems.

You will potentially call the command with different calculator types (e.g. when comparing different methods) and this causes errors, if algorithm in not defined for both of them. You could ignore algorithm all together to solve this. But this then goes on to the question, why it is there in the first place?